packages/headless/src/features/folding/insight-folding-actions.ts

```diff
@@ -0,0 +1,68 @@
+import {createAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {isErrorResponse} from '../../api/search/search-api-client';
+import {AsyncThunkInsightOptions} from '../../api/service/insight/insight-api-client';
+import {validatePayload} from '../../utils/validate-payload';
+import {
+  RegisterFoldingActionCreatorPayload,
+  foldingOptionsSchemaDefinition,
+  LoadCollectionFulfilledReturn,
+  StateNeededByLoadCollection,
+} from '../folding/folding-actions';
+import {ResultWithFolding} from '../folding/folding-slice';
+import {CollectionId} from '../folding/folding-state';
+import {fetchFromAPI} from '../insight-search/insight-search-actions';
+import {buildInsightSearchRequest} from '../insight-search/insight-search-request';
+
+export type {
+  RegisterFoldingActionCreatorPayload,
+  LoadCollectionFulfilledReturn,
+  StateNeededByLoadCollection,
+};
+
+export const registerFolding = createAction(
+  'folding/register',
+  (payload: RegisterFoldingActionCreatorPayload) =>
+    validatePayload(payload, foldingOptionsSchemaDefinition)
+);
+
+// export type StateNeededByLoadCollection = ConfigurationSection &
+//   FoldingSection &
+//   QuerySection &
+//   InsightConfigurationSection;
```

Comment:
Looks like this should be uncommented and used instead of the one from `../folding/folding-actions`.